src/gui/editors/SongEditor.cpp

```diff
@@ -877,17 +832,49 @@ void SongEditor::updatePositionLine()
 
 
 
+//! Convert zoom slider's value to bar width in pixels
+int SongEditor::calculatePixelsPerBar() const
+{
+	// What we need to raise 2 by to get MIN_PIXELS_PER_BAR and MAX_PIXELS_PER_BAR
+	double minExp = log2(MIN_PIXELS_PER_BAR);
+	double maxExp = log2(MAX_PIXELS_PER_BAR);
+	double exponent = m_zoomingModel->value() / double(ZOOM_STEPS) * (maxExp - minExp) + minExp;
+
+	double ppb = pow(2, exponent);
+
+	return static_cast<int>(std::round(ppb));
+}
+
+
+
+
+//! Convert bar width in pixels to zoom slider value
+int SongEditor::calculateZoomSliderValue(int pixelsPerBar) const
+{
+	// What we need to raise 2 by to get MIN_PIXELS_PER_BAR and MAX_PIXELS_PER_BAR
+	double minExp = log2(MIN_PIXELS_PER_BAR);
+	double maxExp = log2(MAX_PIXELS_PER_BAR);
+	double exponent = log2(pixelsPerBar);
+
+	double sliderValue = (exponent - minExp) / (maxExp - minExp) * ZOOM_STEPS;
+
+	return static_cast<int>(std::round(sliderValue));
+}
+
+
+
+
 void SongEditor::zoomingChanged()
 {
-	setPixelsPerBar(static_cast<int>(m_zoomingLogValue / 100.0f * DEFAULT_PIXELS_PER_BAR));
+	int ppb = calculatePixelsPerBar();
+	setPixelsPerBar(ppb);
 
-	m_song->m_playPos[Song::Mode_PlaySong].m_timeLine->
-					setPixelsPerBar( pixelsPerBar() );
+	m_song->m_playPos[Song::Mode_PlaySong].m_timeLine->setPixelsPerBar(ppb);
 	realignTracks();
 	updateRubberband();
 	m_timeLine->setSnapSize(getSnapSize());
 
-	emit zoomingValueChanged(m_zoomingLogValue / 100.0f);
```

Comment:
Since I removed all use of percentage zoom in the calculations, I did that for this signal too